pkg/util/testutil/patternscanner.go

```diff
@@ -15,32 +15,49 @@ import (
 	"testing"
 )
 
+// NoPattern is a sugar syntax for empty pattern
+var NoPattern *regexp.Regexp
+
 // PatternScanner is a helper to scan logs for a given pattern.
 type PatternScanner struct {
-	// The log pattern to match on
-	pattern *regexp.Regexp
+	// The log pattern to match on validating successful start
+	startPattern *regexp.Regexp
+	// The log pattern to match on validating successful finish. This is optional
+	finishPattern *regexp.Regexp
 	// Once we've found the correct log, we should notify the caller.
 	DoneChan chan struct{}
 	// A sync.Once instance to ensure we notify the caller only once, and stop the operation.
 	stopOnce sync.Once
-	// A helper to spare redundant calls to the analyzer once we've found the relevant log.
+
+	// flag to indicate that start was found, and we should look for finishPattern now
+	started bool
```

Comment:
imho, the name is not indicative enough and can be misleading
I interpreted it as "patternScanner started" and not as "start log was found"

Also in the functions themselves it confuses the reading, especially as we have an attribute called "stopped" which indicates the scanner was stopped